### PR TITLE
rename p3_function to p3

### PR DESCRIPTION
**Pull request by xyuan**
rename p3_function to p3

BFB

### CODE REVIEW

**Comment by xyuan** [comment]
It is hard to understand, I use "for filename in p3_functions_*; do git mv "$filename" "${filename//p3_functions_/p3_}"; done" in the batch script to rename the files, but git still shows some files are newly created, not rename, see for example, 

 **create mode** 100644 components/scream/src/physics/p3/p3_incloud_mixingratios.cpp
 **rename components**/scream/src/physics/p3/{p3_functions_incloud_mixingratios_impl.hpp => p3_incloud_mixingratios_impl.hpp} (96%)

File: components/scream/src/physics/p3/p3_autoconversion.cpp [hunk 1]
```diff
@@ -1,4 +1,4 @@
-#include "p3_functions_check_values_impl.hpp"
```
Comment (jeff-cohere) [inline components/scream/src/physics/p3/p3_autoconversion.cpp:1]: Do you know why this was renamed from `..._check_values_impl...` to `..._autoconversion_impl...`?

File: components/scream/src/physics/p3/p3_calc_rime_density.cpp [hunk 2]
```diff
@@ -1,4 +1,4 @@
-#include "p3_functions_conservation_impl.hpp"
```
Comment (jeff-cohere) [inline components/scream/src/physics/p3/p3_calc_rime_density.cpp:1]: Here's another mismatched replacement.

File: components/scream/src/physics/p3/p3_check_values.cpp [hunk 3]
```diff
@@ -1,4 +1,4 @@
-#include "p3_functions_cloud_rain_acc_impl.hpp"
```
Comment (jeff-cohere) [inline components/scream/src/physics/p3/p3_check_values.cpp:1]: Another mismatch.

File: components/scream/src/physics/p3/p3_cldliq_imm_freezing.cpp [hunk 4]
```diff
@@ -1,4 +1,4 @@
-#include "p3_functions_ice_collection_impl.hpp"
```
Comment (jeff-cohere) [inline components/scream/src/physics/p3/p3_cldliq_imm_freezing.cpp:1]: Another mismatch

**Comment by jeff-cohere** [comment]
Okay, it looks like Git is getting confused about which files were being renamed what, because a lot of the comments in the header files with mismatches were copy-pasted from other files and not changed to properly reflect their purpose. This is a weakness with hand-conversion involving tons of boilerplate code, I suppose.

I think we can ignore the mismatches I've flagged. I'll keep the comments up so you can look at them and convince yourself of my explanation above. I'll finish my review, and if (a) these changes build and produce BFB results and (b) I don't see anything else that worries me, I think it's good to go.

**Comment by xyuan** [comment]
It was confusing to me also, since this is done by an bash script, the sed is used to replace these code inside the script. I also observed that “git move” doesn’t keep the right history. But If we are doing this job manually, the history may be better kept. Not sure people is OK with this disturbance of git history, if not, we need go to the manually change this names.

**Comment by PeterCaldwell** [comment]
It looks like tests are passing now, which is the important thing. I think the list of files github thinks are "deleted" is a great opportunity to fix incorrect comments. I just did the first 2 using the github.com 'edit' mode within the file review tab. See autoconversion.cpp, for example. Unfortunately, making this change didn't convince github this is a mv instead of a delete. @xyuan - would you mind cleaning up the ETI comment in the files that github lists as deleted? 

For SHOC, let's make that boilerplate comment totally generic to prevent it from being misleading like it is here. Perhaps use: "Explicit instantiation on Reals using the default device."

**Comment by xyuan** [comment]
Yes, I will do that this morning.

**Comment by xyuan** [comment]
@PeterCaldwell I have done this change, please review

**Pull request merged**